services/web/server/src/simcore_service_webserver/director_v2_core_dynamic_services.py

```diff
@@ -0,0 +1,218 @@
+        "node_uuid": service_uuid,
+        "basepath": f"/x/{service_uuid}",
+        "service_resources": ServiceResourcesDictHelpers.create_jsonable(
+            service_resources
+        ),
+    }
+
+    headers = {
+        "X-Dynamic-Sidecar-Request-DNS": request_dns,
+        "X-Dynamic-Sidecar-Request-Scheme": request_scheme,
+    }
+
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    started_service = await request_director_v2(
+        app,
+        "POST",
+        url=settings.base_url / "dynamic_services",
+        data=data,
+        headers=headers,
+        expected_status=web.HTTPCreated,
+    )
+
+    assert isinstance(started_service, dict)  # nosec
+    return started_service
+
+
+@log_decorator(logger=log)
+async def stop_dynamic_service(
+    app: web.Application, service_uuid: str, save_state: bool = True
+) -> None:
+    # stopping a service can take a lot of time
+    # bumping the stop command timeout to 1 hour
+    # this will allow to sava bigger datasets from the services
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    await request_director_v2(
+        app,
+        "DELETE",
+        url=(settings.base_url / f"dynamic_services/{service_uuid}").update_query(
+            save_state="true" if save_state else "false",
+        ),
+        expected_status=web.HTTPNoContent,
+        timeout=settings.DIRECTOR_V2_STOP_SERVICE_TIMEOUT,
+    )
+
+
+@log_decorator(logger=log)
+async def stop_dynamic_services_in_project(
+    app: web.Application,
+    user_id: Optional[PositiveInt] = None,
+    project_id: Optional[str] = None,
+    save_state: bool = True,
+) -> None:
+    """Stops all services of either project_id or user_id in concurrently"""
+    running_dynamic_services = await get_dynamic_services(
+        app, user_id=user_id, project_id=project_id
+    )
+
+    services_to_stop = [
+        stop_dynamic_service(
+            app=app, service_uuid=service["service_uuid"], save_state=save_state
+        )
+        for service in running_dynamic_services
+    ]
+    await logged_gather(*services_to_stop)
+
+
+# FIXME: ANE please unduplicate the 2 following calls!!!!
+@log_decorator(logger=log)
+async def retrieve(
+    app: web.Application, service_uuid: str, port_keys: list[str]
+) -> DataType:
+    """Pulls data from connections to the dynamic service inputs"""
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    result = await request_director_v2(
+        app,
+        "POST",
+        url=settings.base_url / f"dynamic_services/{service_uuid}:retrieve",
+        data={"port_keys": port_keys},
+        timeout=settings.get_service_retrieve_timeout(),
+    )
+    assert isinstance(result, dict)  # nosec
+    return result
+
+
+@log_decorator(logger=log)
+async def request_retrieve_dyn_service(
+    app: web.Application, service_uuid: str, port_keys: list[str]
+) -> None:
+    # TODO: notice that this function is identical to retrieve except that it does NOT reaise
```

Comment:
ohh my we have two of these!

Reply:
yes, which should I use, which remove?

Reply:
https://github.com/ITISFoundation/osparc-simcore/issues/3191